### PR TITLE
Update binary_sensor.mqtt.markdown for #19195

### PR DESCRIPTION
**Pull request by emontnemery**
**Description:**
Document template variable `entity_id`

**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#19195

## Checklist:

- [x] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release.
- [x] The documentation follows the [standards][standards].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_components/binary_sensor.mqtt.markdown [hunk 1]
```diff
@@ -163,13 +163,24 @@ binary_sensor:
 ```
 {% endraw %}
 
+### {% linkable_title Toggle the binary sensor each time a message is received on state_topic %}
+{% raw %}
+```yaml
+# Example configuration.yaml entry
+binary_sensor:
+  - platform: mqtt
+    state_topic:lab_button/cmnd/POWER"
```
Comment (klaasnicolaas) [inline source/_components/binary_sensor.mqtt.markdown]:
```suggestion
    state_topic: "lab_button/cmnd/POWER"
```

File: source/_components/binary_sensor.mqtt.markdown [hunk 2]
```diff
@@ -163,13 +163,24 @@ binary_sensor:
...
+    value_template:"{%if is_state(entity_id,\"on\")-%}OFF{%-else-%}ON{%-endif%}"}
```
Comment (klaasnicolaas) [inline source/_components/binary_sensor.mqtt.markdown]:
```suggestion
    value_template: "{%if is_state(entity_id,\"on\")-%}OFF{%-else-%}ON{%-endif%}"
```

File: source/_components/binary_sensor.mqtt.markdown [hunk 3]
```diff
@@ -86,7 +86,7 @@ device_class:
   required: false
   type: string
 value_template:
-  description: Defines a [template](/docs/configuration/templating/#processing-incoming-data) to extract a value from the payload.
+  description: Defines a [template](/docs/configuration/templating/#processing-incoming-data) to extract a value from the payload. Available variables: `entity_id`.
```
Comment (klaasnicolaas) [inline source/_components/binary_sensor.mqtt.markdown]:
```suggestion
  description: "Defines a [template](/docs/configuration/templating/#processing-incoming-data) to extract a value from the payload. Available variables: `entity_id`."
```

**Review by klaasnicolaas — CHANGES_REQUESTED** [review]

**Comment by emontnemery** [comment]
@klaasnicolaas Thanks a lot for the review!

**Review by frenck — APPROVED** [review]

**Pull request merged**